google/cloud/documentai_toolbox/wrappers/document_revision.py

```diff
@@ -0,0 +1,346 @@
+# -*- coding: utf-8 -*-
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""Wrappers for Document AI Document type."""
+
+import dataclasses
+import re
+import copy
+
+from typing import List, TypeVar
+
+from google.cloud import documentai
+from google.cloud import storage
+
+from google.cloud.documentai_toolbox.wrappers.entity import Entity
+
+from google.cloud.documentai_toolbox.wrappers.document import Document
+
+_OP_TYPE = documentai.Document.Provenance.OperationType
+
+DocumentWithRevisions = TypeVar("DocumentWithRevisions")
+
+
+def _get_base_and_revision_bytes(output_bucket: str, output_prefix: str) -> List[bytes]:
+    r"""Returns a list of shards as bytes.
+
+    If the filepaths are gs://abc/def/gh/{1,2,3}.json, then output_bucket should be "abc",
+    and output_prefix should be "def/gh".
+
+    Args:
+        output_bucket (str):
+            Required. The name of the output_bucket.
+
+        output_prefix (str):
+            Required. The prefix of the folder where files are excluding the bucket name.
+
+    Returns:
+        List[bytes]:
+            A list of shards as bytes.
+
+    """
+    text = ""
+    base_doc = []
+    revisions_doc = []
+
+    storage_client = storage.Client()
+
+    blob_list = storage_client.list_blobs(output_bucket, prefix=output_prefix)
+    pb = documentai.Document.pb()
```

Comment:
Why are you accessing the `pb()` in the document? Do these doc.dp.dp files work differently to the document.json files?

Reply:
Yes, they're very different and the only way to get a usable Object is by accessing `pb()` and converting the `dp.bp` files to Document object using `pb`

Reply:
Why not just use `documentai.Document.from_json` (https://github.com/googleapis/proto-plus-python/blob/main/proto/message.py#L407) followed by accessing its `text` attribute? (https://github.com/googleapis/google-cloud-python/blob/main/packages/google-cloud-documentai/google/cloud/documentai_v1/types/document.py#L72)

I think these practically do the same thing, but it may be worthwhile to avoid having to access the lower-level protobuf API `pb` and `FromString`.

Reply:
I agree with @dizcology I think that would be possible.

Note: be sure to use the parameter `ignore_unknown_fields` because if the client library lags behind the core `Document` protos, there could be some fields in `Document` output that isn't in the client library yet. This prevents exceptions from being thrown.

```py
document = documentai.Document.from_json(
    blob.download_as_bytes(), ignore_unknown_fields=True
)
```

Reply:
I am not able to use the `document.from_json` since the `db.pb` isn't bytes of a Document object is bytes of the Document message which is different from what `document.from_json` uses. If i try using `Document.from_json` i get this error `UnicodeDecodeError: 'utf-8' codec can't decode byte 0xb3 in position 3: invalid start byte`